pulsar-client-cpp/tests/AuthTokenTest.cc

```diff
@@ -0,0 +1,206 @@
+
+static const std::string tokenPath = "/tmp/pulsar-test-data/certs/token.txt";
+
+static std::string getToken() {
+    std::ifstream file(tokenPath);
+    std::string str((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+    return str;
+}
+
+TEST(AuthPluginToken, testToken) {
+    ClientConfiguration config = ClientConfiguration();
+    std::string token = getToken();
+    AuthenticationPtr auth = pulsar::AuthToken::createWithToken(token);
+
+    ASSERT_TRUE(auth != NULL);
+    ASSERT_EQ(auth->getAuthMethodName(), "token");
+
+    pulsar::AuthenticationDataPtr data;
+    ASSERT_EQ(auth->getAuthData(data), pulsar::ResultOk);
+    ASSERT_EQ(data->hasDataFromCommand(), true);
+    ASSERT_EQ(data->getCommandData(), token);
+    ASSERT_EQ(data->hasDataForTls(), false);
+    ASSERT_EQ(data->hasDataForHttp(), true);
+    ASSERT_EQ(auth.use_count(), 1);
+
+    config.setAuth(auth);
+    Client client(serviceUrl, config);
+
+    std::string topicName = "persistent://private/auth/test-token";
+    std::string subName = "subscription-name";
+    int numOfMessages = 10;
+
+    Producer producer;
+    Result result = client.createProducer(topicName, producer);
+    ASSERT_EQ(ResultOk, result);
+
+    Consumer consumer;
+    result = client.subscribe(topicName, subName, consumer);
+    ASSERT_EQ(ResultOk, result);
+
+    // handling dangling subscriptions
+    consumer.unsubscribe();
+    client.subscribe(topicName, subName, consumer);
+
+    std::string temp = producer.getTopic();
+    ASSERT_EQ(temp, topicName);
+    temp = consumer.getTopic();
+    ASSERT_EQ(temp, topicName);
+    ASSERT_EQ(consumer.getSubscriptionName(), subName);
+
+    // Send Asynchronously
+    std::string prefix = "test-token-message-";
+    for (int i = 0; i < numOfMessages; i++) {
+        std::string messageContent = prefix + boost::lexical_cast<std::string>(i);
+        Message msg = MessageBuilder()
+                          .setContent(messageContent)
+                          .setProperty("msgIndex", boost::lexical_cast<std::string>(i))
+                          .build();
+        producer.sendAsync(msg, NULL);
+        LOG_INFO("sending message " << messageContent);
+    }
+
+    producer.flush();
+
+    Message receivedMsg;
+    for (int i = 0; i < numOfMessages; i++) {
+        Result res = consumer.receive(receivedMsg);
+        ASSERT_EQ(ResultOk, res);
+
+        std::string expectedMessageContent = prefix + boost::lexical_cast<std::string>(i);
+        LOG_INFO("Received Message with [ content - "
+                 << receivedMsg.getDataAsString() << "] [ messageID = " << receivedMsg.getMessageId() << "]");
+        ASSERT_EQ(receivedMsg.getProperty("msgIndex"), boost::lexical_cast<std::string>(i));
+        ASSERT_EQ(expectedMessageContent, receivedMsg.getDataAsString());
+        ASSERT_EQ(ResultOk, consumer.acknowledge(receivedMsg));
+    }
+}
+
+TEST(AuthPluginToken, testTokenWithHttpUrl) {
```

Comment:
This exercises the binary path too. I don't think you need both.

Reply:
It's almost the same, but I wanted to make sure to cover the 2 possible ways to do the lookups.